python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -149,9 +150,17 @@ def evaluate(
 
         batch_count = 0
         df_grouped_stats = pd.DataFrame()
+        # defining variables to compute running mean and variance for t-test computations
+        agg_count, agg_mean, agg_M2 = 0, 0, 0
         for predictions_dict in test_dataset.map(_predict_fn).take(-1):
             predictions_df = pd.DataFrame(predictions_dict)
 
+            # Accumulating statistics for t-test calculation
+            clicked_records = predictions_df[predictions_df[self.feature_config.get_label("node_name")] == 1.0]
+            diff = (clicked_records[RankingConstants.NEW_RANK] - clicked_records[
```

Comment:
@mohazahran Just noticed - We should probably compute the t-test with 1 / rank instead of rank because we are interested in the mean of the reciprocal ranks.